merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/services/ConstraintAction.java

```diff
@@ -217,22 +247,49 @@ public List<ConstraintResult> getViolations(final PlanId planId, final Optional<
 
         ConstraintResult constraintResult = expression.evaluate(preparedResults, environment);
 
-        if (constraintResult.isEmpty()) continue;
+        if (!constraintResult.isEmpty()){
+          constraintResult.constraintName = entry.getValue().name();
+          constraintResult.constraintId = entry.getKey();
+          constraintResult.constraintType = entry.getValue().type();
+          constraintResult.resourceIds = List.copyOf(names);
+        }
 
-        constraintResult.constraintName = entry.getValue().name();
-        constraintResult.constraintId = entry.getKey();
-        constraintResult.constraintType = entry.getValue().type();
-        constraintResult.resourceIds = List.copyOf(names);
+        constraintResultMap.put(entry.getKey(), Failable.of(constraintResult));
```

Comment:
The way it used to work was that the loop would `continue,` and later when we hit `constraintService.createConstraintRuns` at line 281, it would end up storing an empty {} in the database. Personally, I wanted to hang onto these entries in the `constraintResultMap` for debugging `compiledConstraintMap` but still have things work the same way.

In`InsertConstraintRunsAction.java` we are throwing out the constraint that has no violations or is empty and insert `{}`

```
if (results.get(constraint.id()) != null && !results.get(constraint.id()).isEmpty()) {
        statement.setString(4, constraintResultP.unparse(results.get(constraint.id())).toString());
      } else {
        statement.setString(4, "{}");
      }
```